### PR TITLE
Add front to api

### PR DESCRIPTION
**Pull request by Jasopaum**
This change is [Reviewable](https://reviewable.io/reviews/arkhn/warehouse-api/63)

### CODE REVIEW

File: .github/workflows/front.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+          docker tag $IMAGE:latest $IMAGE:$GITHUB_SHA
+          docker tag $IMAGE:latest $GITHUB_REGISTRY/${{ github.repository }}/front-api:$GITHUB_SHA
+          docker tag $IMAGE:latest $GITHUB_REGISTRY/${{ github.repository }}/front-api:latest
+
+      - name: Publish to the docker hub
+        run: |
+          echo "${{ secrets.DOCKER_PASSWORD }}" |  docker login -u ${{ secrets.DOCKER_LOGIN }} --password-stdin
+          docker push $IMAGE:latest
+
+      - name: Publish to Github Packages
```
Comment (simonvadee) [inline .github/workflows/front.yml:65]: don't publish to github

File: .github/workflows/front.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+      - name: Build
+        run: |
+          cd front
+          yarn build
+
+  publish:
+    name: Build and publish
+    env:
+      GITHUB_REGISTRY: docker.pkg.github.com
```
Comment (simonvadee) [inline .github/workflows/front.yml:41]: no need

File: front/package.json [hunk 3]
```diff
@@ -0,0 +1,62 @@
+{
+  "name": "front",
+  "version": "0.1.0",
+  "private": true,
+  "dependencies": {
+    "@apollo/client": "^3.0.0-beta.50",
+    "@apollo/react-hooks": "^3.1.5",
+    "@blueprintjs/core": "^3.28.1",
```
Comment (simonvadee) [inline front/package.json:8]: ça tej

File: front/package.json [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    "@material-ui/core": "^4.10.0",
+    "@material-ui/icons": "^4.9.1",
+    "@material-ui/lab": "^4.0.0-alpha.54",
+    "@testing-library/jest-dom": "^4.2.4",
```
Comment (simonvadee) [inline front/package.json:12]: you have tests ?

File: front/package.json [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    "react-router-dom": "^5.2.0",
+    "react-scripts": "3.4.1",
+    "typescript": "~3.7.2",
+    "use-react-router": "^1.0.7"
+  },
+  "scripts": {
+    "start": "react-scripts start",
+    "build": "react-scripts build",
+    "test": "react-scripts test",
+    "eject": "react-scripts eject",
```
Comment (simonvadee) [inline front/package.json:40]: ça tej

File: front/package.json [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    "react-router": "^5.2.0",
+    "react-router-dom": "^5.2.0",
+    "react-scripts": "3.4.1",
+    "typescript": "~3.7.2",
+    "use-react-router": "^1.0.7"
+  },
+  "scripts": {
+    "start": "react-scripts start",
+    "build": "react-scripts build",
+    "test": "react-scripts test",
```
Comment (simonvadee) [inline front/package.json:39]: tej

File: front/src/App.test.tsx [hunk 7]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
```
Comment (simonvadee) [inline front/src/App.test.tsx:1]: tej

File: front/src/components/appBar/index.tsx [hunk 8]
```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
+import AppBar from '@material-ui/core/AppBar';
+import Toolbar from '@material-ui/core/Toolbar';
+import Typography from '@material-ui/core/Typography';
+import Button from '@material-ui/core/Button';
+
+import useRouter from "use-react-router";
+
+import { AUTH_TOKEN } from '../../constants';
```
Comment (simonvadee) [inline front/src/components/appBar/index.tsx:10]: rename this const, too confusing

File: front/src/components/search/index.tsx [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+  const [apiError, setApiError] = useState(false)
+
+  const getFhirCollections = async () => {
+    try {
+      const response: any = await axios.get(
+        `${FHIR_API_URL}/list-collections`
+      );
+      setFhirCollections(response.data);
+    } catch (err) {
+      // TODO catch error
```
Comment (simonvadee) [inline front/src/components/search/index.tsx:46]: todo ?

File: front/src/components/search/index.tsx [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+      display: 'flex'
+    },
+  }),
+);
+
+const Search = (): React.ReactElement => {
+  const classes = useStyles();
+
+  const [searchParameters, setSearchParameters] = useState('');
+  const [fhirCollections, setFhirCollections] = useState(['Patient', 'Observation', 'Practitioner'] as string[]);
```
Comment (simonvadee) [inline front/src/components/search/index.tsx:34]: define these constants elsewhere

File: front/src/components/search/index.tsx [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+
+  const executeFhirQuery = async () => {
+    const fhirUrl =
+      `${FHIR_API_URL}${selectedCollection ? '/' + selectedCollection : ''}?${searchParameters}`
+
+    try {
+      const response: any = await axios.get(fhirUrl);
+      setFhirBundle(response.data);
+    } catch (err) {
+      // TODO catch error
```
Comment (simonvadee) [inline front/src/components/search/index.tsx:64]: todo ? a nice toaster ?

File: front/src/constants/index.ts [hunk 12]
```diff
@@ -0,0 +1,5 @@
+export const AUTH_TOKEN = 'ARKHN_AUTH_TOKEN';
```
Comment (simonvadee) [inline front/src/constants/index.ts:1]: rename

File: front/src/react-app-env.d.ts [hunk 13]
```diff
@@ -0,0 +1 @@
+/// <reference types="react-scripts" />
```
Comment (simonvadee) [inline front/src/react-app-env.d.ts:1]: tej

File: front/src/serviceWorker.ts [hunk 14]
```diff
@@ -0,0 +1,149 @@
+// This optional code is used to register a service worker.
```
Comment (simonvadee) [inline front/src/serviceWorker.ts:1]: tej

File: front/src/setupTests.ts [hunk 15]
```diff
@@ -0,0 +1,5 @@
+// jest-dom adds custom jest matchers for asserting on DOM nodes.
```
Comment (simonvadee) [inline front/src/setupTests.ts:1]: tej

File: front/src/components/search/index.tsx (hunk 11 above)
Comment (Jasopaum) [reply front/src/components/search/index.tsx:64]: I didn't find a nice toaster, I just have an "Alert" component displayed for now

**Pull request merged**